api/src/main/java/org/openmrs/module/appointments/service/AppointmentsService.java

```diff
@@ -12,47 +12,51 @@
 import java.util.Date;
 import java.util.List;
 
+import static org.openmrs.module.appointments.constants.PrivilegeConstants.MANAGE_APPOINTMENTS;
+import static org.openmrs.module.appointments.constants.PrivilegeConstants.MANAGE_OWN_APPOINTMENTS;
+import static org.openmrs.module.appointments.constants.PrivilegeConstants.VIEW_APPOINTMENTS;
+
 public interface AppointmentsService {
 
     @Transactional
-	@Authorized({"Manage Appointments"})
+    @Authorized({MANAGE_APPOINTMENTS, MANAGE_OWN_APPOINTMENTS})
```

Comment:
My understanding is that if I'm a provider, I should be able to only modify my own appointments. But does this prevent that from happening? I mean if the authorized user is has only the MANAGE_OWN_APPOINTMENTS, they are still able to call this method even when the appointment they are editing is not theirs.

Reply:
Yes, true. It's a common method to create and edit an appointment for any(both manage and own) privileged providers. In that method's implementation, we are validating the users privileges to take decision on further action. We did not create a separate method for this as that would have been two methods for each action with duplicate code.